FWCore/Concurrency/src/SerialTaskQueue.cc

```diff
@@ -28,15 +28,40 @@ SerialTaskQueue::~SerialTaskQueue() {
   bool isEmpty = m_tasks.empty();
   bool isTaskChosen = m_taskChosen;
   if ((not isEmpty and not isPaused()) or isTaskChosen) {
-    pushAndWait([]() { return; });
+    tbb::task_group g;
+    std::atomic<bool> done = false;
+    push(g, [&done]() {
+      done = true;
+      return;
+    });
+    do {
+      g.wait();
+    } while (not done.load());
```

Comment:
It is my intention to do so as that wait actually doesn't work since there are no running tasks for the newly created group :(. This should be fixed as part of this pull request.